### ISSUE TITLE
Auto-shutdown is not triggered when using sky launch from a CI server

### ISSUE THREAD

**Issue by bilelomrani1** [skypilot-org/skypilot#2115]
A small weird and low priority issue that I discovered during my testing of the features proposed in #2109 is that the auto-shutdown does not seem to work when set from a CI runner. I cannot really explain this yet, and did not investigate it further because using `sky stop` at the end of the job works well for me in this case. Quickly looking at the logs does not reveal anything suspicious. I don't know if this is an artefact of our environment or if there is something deeper happening from SkyPilot.

To replicate, here is a example Gitlab CI job configuration:

```yaml
GPU tests:
  image: google/cloud-sdk:latest
  stage: Test
  before_script:
    - apt-get install rsync -y
    - pip install --upgrade pip
    - pip install skypilot[gcp]
    - echo $GPU_TESTING_SERVICE_ACCOUNT_KEY > gpu-testing-service-account-key.json
    - export GOOGLE_APPLICATION_CREDENTIALS="gpu-testing-service-account-key.json"
    - gcloud auth activate-service-account --key-file gpu-testing-service-account-key.json 
    - gcloud config set project $GCP_PROJECT_ID
    # Hack to bypass SkyPilot hardcoded verification
    # See https://github.com/skypilot-org/skypilot/issues/2109
    - touch ~/.config/gcloud/application_default_credentials.json
  script:
    - sky launch -i 5 -y -c e2e-test-gpu tests/gpu-launcher/launch-gpu-tests.yml
```

The job exits gracefully, but the machine is not stopped even after waiting for more than 30 minutes.

If you fail to replicate and if you need me to investigate something in particular related to this issue, don't hesitate to reach out!

**Comment by Michaelvll** [comment]
That is a great catch! My current guess is the way we handle the autostop causes the service account to fail to terminate the cluster on the head node. Let me try to reproduce it and get back to you soon.

**Comment by Michaelvll** [comment]
Just submitted a PR #2116 for fixing this issue. After that PR, your original scripts should work and the `touch ~/.config/gcloud/application_default_credentials.json` will not be needed anymore. ; )

**Comment by Michaelvll** [comment]
We just fixed this issue #2116, the problem should be fixed now. Please feel free to have a try. : )

**Comment by bilelomrani1** [comment]
Hi @Michaelvll I just tested it and it indeed solved the issue, thank you for your help.
